api/cpp/CMakeLists.txt

```diff
@@ -382,7 +377,7 @@ if(BUILD_TESTING)
         target_compile_definitions(test_${NAME} PRIVATE
             SOURCE_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}/\"
         )
-        add_test(test_${NAME} ${CMAKE_CURRENT_BINARY_DIR}/test_${NAME})
+        add_test(NAME test_${NAME} COMMAND test_${NAME})
```

Comment:
Yes, it is safe that it will be found on the system, because we're passing a CMake target here, and this variant of `add_test` supports that, so it'll fetch the location actually from the cmake target, not assume that 'test_${NAME}' is in `PATH` or so.